app/src/main/java/com/example/iou/map/activities/GoogleMapsActivity.java

```diff
@@ -0,0 +1,187 @@
+package com.example.iou.map.activities;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.FragmentActivity;
+
+import com.codepath.asynchttpclient.AsyncHttpClient;
+import com.codepath.asynchttpclient.callback.JsonHttpResponseHandler;
+import com.example.iou.R;
+import com.example.iou.databinding.ActivityGoogleMapsBinding;
+import com.example.iou.map.models.RestaurantItem;
+import com.google.android.gms.maps.CameraUpdate;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.BitmapDescriptor;
+import com.google.android.gms.maps.model.BitmapDescriptorFactory;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GoogleMapsActivity extends FragmentActivity implements OnMapReadyCallback {
+
+    public String URL;
+    private GoogleMap mMap;
+    private ActivityGoogleMapsBinding binding;
+    private LocationManager locationManager;
+    private final static String KEY_LOCATION = "location";
+    private double longitude;
+    private double latitude;
+    private double restaurantLatitude;
+    private double restaurantLongitude;
+    private List<RestaurantItem> nearbyRestaurants;
+    Location mCurrentLocation;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```

Comment:
This function is very long & the code inside gets pretty deeply nested. Would recommend splitting out some of the code in the listeners into separate functions!